snake.asm

```diff
@@ -30,14 +27,15 @@ sub di, bx              ; Update snake's head position based on movement
 cmp di, cx              ; Check if snake's head position is beyond screen boundaries
 ja start                ; If out of bounds, restart the game
 ; The comparison is unsigned (ja) to determine if snake's head has moved beyond screen boundaries
-; In unsigned comparisons, only the magnitude of the values is considered, rather their signs
+; In unsigned comparison every negative value will be bigger than every positive numbet as the sign bit 
+; is considered as a power of two
 ; this is crucial, because memory addresses like di are treated as unsigned, so the comparison
 ; ensures that if di > cx (screen size) or di < 0, the jump will happen, preventing the snake from
 ; moving outside the screen area.
 sar bx, 0x1             ; Shift the value in bx right by 1 bit (right arithmetic shift)

```

Comment:
Note the possible bx values to make the next comment more clear (as in the original comment)